R/snapshot.R

```diff
@@ -885,40 +885,40 @@ renv_snapshot_filter <- function(project, records, type, packages, exclude) {
   type <- aliases[[type]] %||% type
 
   result <- switch(type,
-    all      = renv_snapshot_filter_all(project, records),
-    custom   = renv_snapshot_filter_custom(project, records),
-    explicit = renv_snapshot_filter_explicit(project, records),
-    implicit = renv_snapshot_filter_implicit(project, records),
-    packages = renv_snapshot_filter_packages(project, records, packages),
+    all      = renv_snapshot_filter_all(project, records, exclude),
+    custom   = renv_snapshot_filter_custom(project, records, exclude),
+    explicit = renv_snapshot_filter_explicit(project, records, exclude),
+    implicit = renv_snapshot_filter_implicit(project, records, exclude),
+    packages = renv_snapshot_filter_packages(project, records, packages, exclude),
     stopf("unknown snapshot type '%s'", type)
   )
 
-  if (length(exclude))
```

Comment:
The PR effectively moves this piece into each of the filter routines, so that filters are processed at the right time.